src/serde_impl.rs

```diff
@@ -102,7 +102,9 @@ macro_rules! impl_serde_for_symbol {
                 deserializer: D,
             ) -> ::core::result::Result<Self, D::Error> {
                 let index = <$ty as ::serde::Deserialize<'de>>::deserialize(deserializer)?;
-                let ::core::option::Option::Some(symbol) = Self::new(index) else {
+                let ::core::option::Option::Some(symbol) =
+                    index.try_into().ok().and_then(Self::try_from_usize)
+                else {
```

Comment:
This is unnecessary since `Self::new` already does what was needed.
Please correct me if my assessment here is wrong.

Reply:
Just revert those changes.

Reply:
Sure, Self::new() is what does the work so the extra level of indirection isn't necessary. In my eyes it's just nice to match the usage of to_usize() in the serialize impl, but it's matter of taste.

Anyway, I've reverted this change now as requested.